lib/generator/files/exercise_files.rb

```diff
@@ -0,0 +1,83 @@
+require 'erb'
+
+module Generator
+  module Files
+    module GeneratorCases
+      def self.available(track_path)
+        generator_glob = File.join(track_path, 'lib', '*_cases.rb')
+        Dir[generator_glob].map { |filename| exercise_name(filename) }
+      end
+
+      def self.filename(exercise_name)
+        "#{exercise_name.tr('-','_')}_cases"
+      end
+
+      def self.proc_name(exercise_name)
+        filename(exercise_name).split('_').map(&:capitalize).join
+      end
+
+      private
+
+      def self.exercise_name(filename)
+        %r{([^/]*)_cases\.rb$}.match(filename).captures[0].tr('_','-')
+      end
+
+    end
+
+    # Assumes methods exist: paths, exercise_name
```

Comment:
Excellent point. Thanks.

This sort of thing is what I love about code reviews. :heart: